### PR TITLE
#2404 - Save Name and Reaction Conditions to RDF Metadata

### PR DESCRIPTION
**Pull request by even1024**
## Generic request
- [x] PR name follows the pattern `#1234 – issue name`
- [x] branch name does not contain '#'
- [x] base branch (master or release/xx) is correct
- [x] PR is linked with the issue
- [x] task status changed to "Code review"
- [x] code follows product standards
- [x] regression tests updated

### CODE REVIEW

File: core/indigo-core/layout/src/pathway_layout.cpp [hunk 1]
```diff
@@ -341,13 +350,19 @@ void PathwayLayout::generateTextBlocks(SimpleTextObjectBuilder& tob, const ObjAr
 {
     for (int i = 0; i < props.size(); ++i)
     {
-        if (height > _text_height)
+        if (std::round(height * 1000) >= std::round(_text_line_height * 1000))
```
Comment (AlexanderSavelyev) [inline core/indigo-core/layout/src/pathway_layout.cpp:353]: add constant for 1000

File: core/indigo-core/layout/src/pathway_layout.cpp [hunk 2]
```diff
@@ -341,13 +350,19 @@ void PathwayLayout::generateTextBlocks(SimpleTextObjectBuilder& tob, const ObjAr
...
         {
-            height -= _text_height;
+            height -= _text_line_height;
             SimpleTextLine textLine;
             textLine.text = props[i].ptr();
-            if (height < _text_height && props.size() - i > 1)
-                textLine.text += "...";
+            if (std::round(height * 1000) < std::round(_text_line_height * 1000) && props.size() - i > 1)
```
Comment (AlexanderSavelyev) [inline core/indigo-core/layout/src/pathway_layout.cpp:358]: add constant for 1000

**Review by AlexanderSavelyev — CHANGES_REQUESTED** [review]

File: core/indigo-core/layout/src/pathway_layout.cpp (hunk 2 above)
Comment (even1024) [reply core/indigo-core/layout/src/pathway_layout.cpp]: fixed

File: core/indigo-core/layout/src/pathway_layout.cpp (hunk 1 above)
Comment (even1024) [reply core/indigo-core/layout/src/pathway_layout.cpp]: fixed

**Review by AlexanderSavelyev — APPROVED** [review]

**Pull request merged**